iquip/apps/dataviewer.py

```diff
@@ -156,3 +161,80 @@ def setData(self, data: np.ndarray, axes: Sequence[AxisInfo]):
         x, y = haxis.values[0], vaxis.values[0]
         width, height = haxis.values[-1] - x, vaxis.values[-1] - y
         self.image.setRect(x, y, width, height)
+
+
+class _RealtimePart(QWidget):
+    """Part widget for configuring realtime mode of the source widget.
+    
+    Attributes:
+        label: Label for showing information about the current experiment.
+          When it is synchronized with an experiment, it displays the RID
+          of the experiment. Otherwise, it shows "No running experiment.".
+        button: Button for synchronizing with the current artiq master.
+    """
+
+    def __init__(self, parent=None):
+        """Extended."""
+        super().__init__(parent=parent)
+        self.label = QLabel("No running experiment.", self)
+        self.button = QPushButton("Sync", self)
+        layout = QHBoxLayout(self)
+        layout.addWidget(self.label)
+        layout.addWidget(self.button)
+
+
+class _RemotePart(QWidget):
+    """Part widget for configuring remote mode of the source widget.
+    
+    Attributes:
+        spinbox: Spinbox for RID input.
+        label: Label for showing the execution time of the experiment.
+    """
+
+    def __init__(self, parent=None):
+        """Extended."""
+        super().__init__(parent=parent)
+        self.spinbox = QSpinBox(self)
+        self.spinbox.setRange(0, 999_999_999)
```

Comment:
Setting the maximum value to just a random big number looks not good.

How about setting it to the maximum value of `int` type?
The max value of spin box is limitted to that of `int` type, so it looks more reasonable!

Additionally, the default minimum value is 0, and you don't need to set it.

Reply:
Okay then I will change it to `setMaximum(2**31 - 1)`.